static/app/components/smartSearchBar/index.tsx

```diff
@@ -749,13 +769,222 @@ class SmartSearchBar extends React.Component<Props, State> {
+  ): SearchItem[] {
+    const operatorMap = {
+      [TermOperator.Default]: {
+        type: 'tag-operator' as ItemType,
+        value: ':',
+        desc: t(`${tag}:[value] is equal to`),
+      },
+      [TermOperator.GreaterThanEqual]: {
+        type: 'tag-operator' as ItemType,
+        value: ':>=',
+        desc: t(`${tag}:>=[value] is greater than or equal to`),
+      },
+      [TermOperator.LessThanEqual]: {
+        type: 'tag-operator' as ItemType,
+        value: ':<=',
+        desc: t(`${tag}:<=[value] is less than or equal to`),
+      },
+      [TermOperator.GreaterThan]: {
+        type: 'tag-operator' as ItemType,
+        value: ':>',
+        desc: t(`${tag}:>[value] is greater than`),
+      },
+      [TermOperator.LessThan]: {
+        type: 'tag-operator' as ItemType,
+        value: ':<',
+        desc: t(`${tag}:<[value] is less than`),
+      },
+      [TermOperator.Equal]: {
+        type: 'tag-operator' as ItemType,
+        value: ':=',
+        desc: t(`${tag}:=[value] is equal to`),
+      },
+      [TermOperator.NotEqual]: {
+        type: 'tag-operator' as ItemType,
+        value: '!:',
+        desc: t(`!${tag}:[value] is not equal to`),
+      },
+    };
+    return validOps.map(op => operatorMap[op]);
+  }
+
+  getValidOps(filterToken: TokenResult<Token.Filter>): readonly TermOperator[] {
+    const types = filterToken.invalid?.expectedType ?? [filterToken.filter];
+    const filterType = interchangeableFilterOperators[types[0]]
+      ? interchangeableFilterOperators[types[0]][0]
+      : types[0];
+    const config = filterTypeConfig[filterType];
+    return config.validOps ?? [''];
+  }
+
+  generateOpAutocompleteGroup(
+    validOps: readonly TermOperator[],
+    tagName: string
+  ): AutocompleteGroup {
+    const operatorItems = this.generateOpAutocompleteEntries(validOps, tagName);
+    return {
+      searchItems: operatorItems,
+      recentSearchItems: undefined,
+      tagName: '',
+      type: 'tag-operator' as ItemType,
+    };
+  }
+
+  withinTokenLocation(node: {location: LocationRange}, position: number) {
+    return position >= node.location.start.offset && position <= node.location.end.offset;
+  }
+
+  getCursorToken = (AST: ParseResult, cursor: number): ParseResult[number] | null => {
+    // traverse AST to find first matching filter based on cursor position
+    for (const node of AST) {
+      if (node !== Token.Spaces && this.withinTokenLocation(node, cursor)) {
+        // traverse into a logic group to find specific filter
+        if (node.type === Token.LogicGroup) {
+          return this.getCursorToken(node.inner, cursor);
+        }
+        return node;
+      }
+    }
+    return null;
+  };
+
   updateAutoCompleteItems = async () => {
     if (this.blurTimeout) {
       clearTimeout(this.blurTimeout);
       this.blurTimeout = undefined;
     }
 
     const cursor = this.getCursorPosition();
+    const {organization} = this.props;
+    const AST = this.state.parsedQuery;
```

Comment:
kinda a `nit`, but I would avoid uppercase here, since usually a variable starting with uppercase is reserved for either React components or classes.

I think `ast` would be OK here` Or even just pull it out of state as 

```tsx
const {parsedQuery} = this.state;
```

Reply:
will change to just using `parsedQuery`